packages/server/graphql/intranetSchema/mutations/updateEmail.ts

```diff
@@ -0,0 +1,74 @@
+import {GraphQLBoolean, GraphQLNonNull, GraphQLString} from 'graphql'
+import getRethink from '../../../database/rethinkDriver'
+import {requireSU} from '../../../utils/authorization'
+import {InternalContext} from '../../graphql'
+import updateUser from '../../../postgres/queries/updateUser'
+
+const updateEmail = {
+  type: GraphQLNonNull(GraphQLBoolean),
+  description: `Updates the user email`,
+  args: {
+    oldEmail: {
+      type: GraphQLNonNull(GraphQLString),
+      description: 'User current email'
+    },
+    newEmail: {
+      type: GraphQLNonNull(GraphQLString),
+      description: 'User new email'
+    }
+  },
+  resolve: async (_source, {oldEmail, newEmail}, {authToken}: InternalContext) => {
+    const r = await getRethink()
+
+    // AUTH
+    requireSU(authToken)
+
+    // VALIDATION
+    if (!newEmail) {
+      throw new Error('New email can not be empty')
+    }
+
+    if (oldEmail === newEmail) {
+      throw new Error('New email is the same as the old one')
+    }
+
+    const email = newEmail.toLowerCase().trim()
+    if (!email.includes('@')) {
+      throw new Error('Invalid email')
+    }
+
+    const user = await r
+      .table('User')
+      .getAll(oldEmail, {index: 'email'})
+      .nth(0)
+      .default(null)
+      .run()
+    if (!user) {
+      throw new Error(`User with ${oldEmail} not found`)
+    }
+
+    // RESOLUTION
+    const userId = user.id
```

Comment:
this is gonna be the absolutely most nittiest of requests & if you don't like it i'm totally OK with that :see_no_evil: 
when possible, could you destructure like `const {id: userId} = user`? 
Even though it's a little bit longer, it means fewer line diffs so in the future if we need to get more props out of that object e.g. `const {id: userId, createdAt, updatedAt} = user` is just gonna look like a partial line diff

Reply:
I actually think that's a very valid request, fixed 😅